### PR TITLE
Adds user signup

### PR DESCRIPTION
**Pull request by jaenia**
[As a User I can signup to an account](https://trello.com/c/pD9jTQgy/10-as-a-user-i-can-signup-to-an-account)

### Description
This PR adds feature of signup on PokeBattle. Next steps are include the possibility to login and logout, and uses logged user as creator os battles.

### How to test
- [ ] Access `localhost:8000/user/signup`
- [ ] Insert a user and a password. You'll be redirected to battles list, where you can see your email beside the header of the page.

### CODE REVIEW

File: backend/battles/views.py [hunk 1]
```diff
@@ -39,6 +39,11 @@ class BattleDetail(DetailView):
 class BattleList(ListView):
     model = Battle
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["logged_user"] = self.request.user
```
Comment (filipeximenes) [inline backend/battles/views.py]: No need for this. The logged user is automatically included in the templates: https://docs.djangoproject.com/en/3.1/ref/templates/api/#django-contrib-auth-context-processors-auth

So is the request object: https://docs.djangoproject.com/en/3.1/ref/templates/api/#django-template-context-processors-request (so you could do `request.user` in any case)

**Review by filipeximenes — CHANGES_REQUESTED** [review]
Good job! One comment

File: backend/battles/templates/battles/battle_list.html [hunk 2]
```diff
@@ -1,9 +1,12 @@
 {% load static %}
 
-<link rel="stylesheet" type="text/css" href="{% static 'css/styles.css' %}" />
 <link rel="stylesheet" type="text/css" href="{% static 'css/global.css' %}" />
+<link rel="stylesheet" type="text/css" href="{% static 'css/battle_styles.css' %}" />
 
-<h1>Battles</h1>
+<div class="header">
+  <h1>Battles</h1>
+  <p>{{ request.user }}</p>
```
Comment (filipeximenes) [inline backend/battles/templates/battles/battle_list.html:8]: it could be just `{{ user }}``

**Review by filipeximenes — APPROVED** [review]

File: backend/battles/templates/battles/battle_list.html (hunk 2 above)
Comment (filipeximenes) [reply backend/battles/templates/battles/battle_list.html:8]: Oh.. and in general, you don't want to "print" the object directly. It's better to be more specific about what you want to display. Eg.: `user.get_full_name()`

Comment (jaenia) [reply backend/battles/templates/battles/battle_list.html:8]: Will do in a future PR.

**Pull request merged**